warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctionsDescriptor.java

```diff
@@ -110,28 +115,55 @@ public Set<String> fields(MetadataHelper helper, Set<String> datatypeFilter) {
                     fields.addAll(JexlASTHelper.getIdentifierNames(functionMetadata.args().get(i)));
                 }
             }
-            return fields;
+            
+            for (String field : fields) {
+                filterField(allFields, field, filteredFields);
+            }
```

Comment:
I would update filterField to accept the set of fields, and remove the for loop here since that's how it's being used throughout this class.

Reply:
Updated